xcodeproj/internal/xcode_targets.bzl

```diff
@@ -307,26 +353,35 @@ def _merge_xcode_target(*, src, dest):
         extensions = dest._extensions,
         app_clips = dest._app_clips,
         dependencies = depset(
-            transitive = [dest._dependencies, src._dependencies],
+            transitive = [dest._dependencies] + [src._dependencies for src in [src_swift, src_other] if src],
```

Comment:
Put somewhere by `srcs = []`:
```
transitive_dependencies = [dest._dependencies]
```

Then inside of `if src_swift:`
```
    transitive_dependencies.append(src_swift._dependencies)
```
and the same for `src_other`.